### PR TITLE
 Issue 4490: Improve error logging on client

### LINKED ISSUE #4490: Improve error logging on client
**Problem description**
To aid in debugging problems in Pravega we should make it easier to track errors across multiple processes. So when an error comes back from the controller or the segment store, we should be sure to log:
 * The request id
 * The appropriate data/stack trace
 * Which server in came from (ip or hostname) so that what happened can be easily located in its log.

**Problem location**
`ControllerImpl`, `SegmentOutputStreamImpl`, `AsyncSegmentInputStreamImpl` and `RawClient`

**Suggestions for an improvement**
Check each of these 4 classes and inspect each location where something is logged at `warn` or `error` level and make sure enough context is being included that the problem could be located, even if it is in the logs of another host.

### PR DESCRIPTION
**Pull request by zwx14700**
Signed-off-by: Wenxiao Zhang <wenxiao.zhang@emc.com>

**Change log description**  
Adding more information in client side logging.

**Purpose of the change**  
Fixed #4490 

**What the code does**  
No changes to functionality. Improving the logging.

**How to verify it**  
All the existing test should continue to pass.

### CODE REVIEW

File: client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java [hunk 1]
```diff
@@ -757,7 +760,7 @@ private Boolean handleScaleResponse(Stream stream, ScaleResponse response, long
 
         return getSegmentsBetweenStreamCuts(from, StreamCut.UNBOUNDED).whenComplete((x, e) -> {
             if (e != null) {
-                log.warn("getSuccessors failed: ", e);
+                log.warn("getSuccessorsFromCut failed: ", e);
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java]: This should include the segment.

File: client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java [hunk 2]
```diff
@@ -776,7 +779,7 @@ private Boolean handleScaleResponse(Stream stream, ScaleResponse response, long
 
         return getSegmentsBetweenStreamCuts(fromStreamCut, toStreamCut).whenComplete((x, e) -> {
             if (e != null) {
-                log.warn("getSuccessors failed: ", e);
+                log.warn("getSegments failed: ", e);
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java]: This should include the stream name

File: client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java [hunk 3]
```diff
@@ -946,7 +949,7 @@ private TxnSegments convert(CreateTxnResponse response) {
             }
         }).whenComplete((s, e) -> {
             if (e != null) {
-                log.warn("Ping Transaction failed:", e);
+                log.warn("PingTransaction failed:", e);
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java]: Can this include the transactionId?

File: client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java (hunk 1 above)
Comment (zwx14700) [reply client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java]: Fixed.

File: client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java (hunk 2 above)
Comment (zwx14700) [reply client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java]: Fixed.

File: client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java (hunk 3 above)
Comment (zwx14700) [reply client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java]: fixed.

**Comment by derekm** [comment]
I saw an issue related to this PR here:
https://github.com/pravega/pravega/blame/master/client/src/main/java/io/pravega/client/segment/impl/AsyncSegmentInputStreamImpl.java#L196
where `connection1` is `null` in the log output.

See https://travis-ci.org/github/pravega/pravega/jobs/714620050#L8352
```
2020-08-03 22:14:09,021 167702 [clientInternal-22-2] WARN  i.p.c.s.i.AsyncSegmentInputStreamImpl - Exception while establishing connection with Pravega node null: 
java.util.concurrent.CompletionException: io.pravega.shared.protocol.netty.ConnectionFailedException: java.net.ConnectException: Connection refused (Connection refused)
```

**Pull request merged**